cmd/arcaflow/main.go

```diff
@@ -139,7 +139,7 @@ Options:
 
 	var configData any = map[any]any{}
 	if configFile != "" {
-		configData, err = loadYamlFile(requiredFilesAbsPaths[RequiredFileKeyConfig])
+		configData, err = loadYamlFile(*fileCtx.AbsPathByKey(RequiredFileKeyConfig))
```

Comment:
Come to think of it, this makes the cached byte stream seem a bit silly since you're not using it here, just grabbing the absolute path and passing to a function that (re)reads the file. Don't you really just want to "unmarshall" the YAML from the file bytestream you already cached? (Forget I said that, however, since it's probably outside any reasonable scope for this PR!)

Reply:
you're right, I saw that last time I worked on this. It needs to get changed.

Reply:
`NewFileCacheUsingContext()` doesn't actually read the file(s), so that work isn't wasted here....

Reply:
Right. `NewFileCacheUsingContext` is the one that doesn't have a "file cache". Yeah, OK: too many similar overlapping maps with confusing names to track. I'm not crazy about that. I give up. (For today: I'll look again tomorrow.)